src/ValueHandler/MetricPropertyValueHandler.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+
+namespace Webgriffe\SyliusAkeneoPlugin\ValueHandler;
+
+
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Webgriffe\SyliusAkeneoPlugin\Converter\DefaultUnitMeasurementValueConverterInterface;
+use Webgriffe\SyliusAkeneoPlugin\ValueHandlerInterface;
+use Webmozart\Assert\Assert;
+
+final class MetricPropertyValueHandler implements ValueHandlerInterface
+{
+    /** @var PropertyAccessorInterface */
+    private $propertyAccessor;
+
+    /** @var string */
+    private $akeneoAttributeCode;
+
+    /** @var string */
+    private $propertyPath;
+
+    /** @var string|null */
+    private $defaultAkeneoUnitMeasurementCode;
+    /**
+     * @var DefaultUnitMeasurementValueConverterInterface
+     */
+    private $defaultUnitMeasurementValueConverter;
+
+    public function __construct(
+        PropertyAccessorInterface $propertyAccessor,
+        DefaultUnitMeasurementValueConverterInterface $defaultUnitMeasurementValueConverter,
+        string $akeneoAttributeCode,
+        string $propertyPath,
+        string $defaultAkeneoUnitMeasurementCode = null
+    ) {
+        $this->propertyAccessor = $propertyAccessor;
+        $this->defaultUnitMeasurementValueConverter = $defaultUnitMeasurementValueConverter;
+        $this->akeneoAttributeCode = $akeneoAttributeCode;
+        $this->propertyPath = $propertyPath;
+        $this->defaultAkeneoUnitMeasurementCode = $defaultAkeneoUnitMeasurementCode;
+    }
+
+    /**
+     * @param mixed $subject
+     */
+    public function supports($subject, string $attribute, array $value): bool
+    {
+        return $subject instanceof ProductVariantInterface &&
+            $attribute === $this->akeneoAttributeCode &&
+            array_key_exists('0', $value) &&
+            is_array($value[0]) &&
+            array_key_exists('data', $value[0]) &&
+            is_array($value[0]['data']) &&
+            array_key_exists('amount', $value[0]['data']) &&
+            array_key_exists('unit', $value[0]['data'])
```

Comment:
In all other value handlers there aren't these checks. We're hiding an issue this way. If an Akeneo attribute is configured with a metric attribute value handler its value must be a metric value.